baseplate/frameworks/thrift/__init__.py

```diff
@@ -64,11 +83,40 @@ def __init__(
         context: RequestContext,
         logger: Logger,
         convert_to_baseplate_error: bool,
+        tracer: Tracer,
     ):
         self.handler = handler
         self.context = context
         self.logger = logger
         self.convert_to_baseplate_error = convert_to_baseplate_error
+        self._tracer = tracer
+
+    @contextmanager
+    def _set_remote_context(self, request_context: RequestContext) -> Iterator[None]:
+        headers = request_context.headers
+        if headers:
+            header_dict = {}
+            for k, v in headers.items():
+                try:
+                    header_dict[k.decode()] = v.decode()
+                except UnicodeDecodeError:
+                    self.logger.info("Unable to decode header %s, ignoring." % k.decode())
+
+            ctx = propagator.extract(header_dict)
+            logger.debug("Extracted trace headers. [ctx=%s, header_dict=%s]" % (ctx, header_dict))
```

Comment:
🔕 I believe this formats the log message even if debug logging is disabled.
```suggestion
            logger.debug("Extracted trace headers. [ctx=%s, header_dict=%s]", ctx, header_dict)
```